File: python/7 kyu/132.py
# https://www.codewars.com/kata/6885f0a045008e707baa714b/train/python

# Passed

def read(tape, head, moves):
    indexes_to_read = []
    for move in moves:
        indexes_to_read.append(head)

        if move == ">":
            head += 1
        elif move == "<":
            head -= 1
        
    result = "".join([tape[idx] for idx in indexes_to_read])
    return result
        
output = read('011010', 3, '>><<')
print(output)

File: nodejs/7 kyu/132.js
// https://www.codewars.com/kata/58f8b35fda19c0c79400020f/train/javascript

// Passed

function allNonConsecutive(arr) {
  let nonConsecutives = [];
  for (let idx = 0; idx < arr.length; idx++) {
    if (idx == arr.length - 1) continue;

    if (arr[idx + 1] - arr[idx] !== 1) {
      // found a non consecutive at idx + 1
      let nonConsecutive = arr[idx + 1];
      nonConsecutives.push({
        i: idx + 1,
        n: nonConsecutive,
      });
    }
  }

  return nonConsecutives;
}

const output = allNonConsecutive([1, 2, 3, 4, 6, 7, 8, 10]);
console.log(output);